### PR TITLE
Fix issues when used within ember-engines.

### LINKED ISSUE #7: Doesn't work in an engine?
I'm not sure this is the right repo to file a bug, but I think this addon is broken when used in an engine. I've got an unusual setup though, so it might be something else.

Basically when the dynamic element is invoked (it's being called from ember-basic-dropdown) I get `Error: Could not find module '<my engine name>/components/-dynamic-element' imported from '(require)'`. 

Following the stack trace up, it looks like the engine file has the line `define.alias("@ember/component", "<my engine name>/components/-dynamic-element");`. When `findModule` runs, it tried to resolve `@ember/component`, but that isn't in the registry.

This is using ember and ember-cli 3.11.

### PR DESCRIPTION
**Pull request by rwjblue**
Avoiding reexport directly here because in some circumstances (ember-engines for example) a simple reexport is transformed to `define.alias`, unfortunately at the moment (ember-source@3.13) there is no _actual_ `@ember/component` module to alias so this causes issues

tldr; we can replace this with a simple reexport when we can rely on Ember actually providing a `@ember/component` module

Fixes #7
Fixes https://github.com/ember-engines/ember-engines/issues/676

### CODE REVIEW

**Comment by chancancode** [comment]
Is this a bug with the babel plugin?

**Comment by rwjblue** [comment]
@chancancode its an ordering issue, we have two babel plugins that run each with very specific responsibilities:

1. replace reexport only modules with `define.alias` ([docs here](https://github.com/ember-cli/loader.js/#definealiasoldpath-new-name))
2. replace any usage of Ember's modules API with the globals 

https://github.com/ember-cli/babel-plugin-ember-modules-api-polyfill/pull/31 and https://github.com/ember-cli/babel-plugin-ember-modules-api-polyfill/pull/32 fixed babel-plugin-ember-modules-api-polyfill to deal with reexports (by transpiling them to `export default Ember.Component`), but the way that the modules polyfill is applied to `ember-cli-babel`'s `transpileTree` ([docs here](https://github.com/babel/ember-cli-babel#additional-trees)) means that it runs _after_ the `babel-plugin-compact-reexports` (which is added by ember-engines [here](https://github.com/ember-engines/ember-engines/blob/84b4b2da762a5ca67058a4cc5566bce6664f479a/lib/engine-addon.js#L489)).

tldr; sure, its probably a bug somewhere, this is probably the easiest fix as far as I can tell

**Comment by jamescdavis** [comment]
Any chance of getting this merged and into a release? We've been on Rob's fork's fix for some time and would love to move back to an official release.

**Comment by lougreenwood** [comment]
👍 on a release, this is also blocking us too 👼

**Comment by chancancode** [comment]
Sorry it took so long! It's published as 0.2.0
